### PR TITLE
Feat/deliverable3

### PR DESCRIPTION
**Pull request by laulujan**
## Description

Deliverable 3
This PR includes:

- Routing (instead of conditional rendering).
- Fetching Banners, products, categories and search from API.
- Pagination
- Product details view
- Search results view

## Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [x] Refactor (non-breaking refactor which doesn't change functionality)
- [x] Styles (Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc))

## How should this be tested?

manual exploratory testing

## What is the deliverable?
Deliverable 3: Fetching a Real API, React Router, Global State Management
## Screenshots and links
[Live version](https://sharp-varahamihira-e41901.netlify.app/home)
Screen Shot 2022-02-21 at 10 05 31

## Checklist
- [ ] I added the necessary documentation, if appropriate.
- [ ] I added tests to prove that my fix is effective or my deliverable works.
- [x] I reviewed existing Pull Requests before submitting mine.
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/api/fetchData.js [hunk 1]
```diff
@@ -1,3 +1,19 @@
+import { API_BASE_URL } from '../utils/constants';
```
Comment (ragrum) [inline src/api/fetchData.js]: As we talked you're not using this function or file at all, please remove it.

File: src/components/Pagination/Pagination.jsx [hunk 2]
```diff
@@ -6,17 +6,40 @@ import {
...
   RightIcon,
 } from './Pagination.styled';
+import { useProducts } from '../../provider/Provider';
 
-const Pagination = () => {
+const Pagination = ({ page }) => {
+  const { totalPages, updatePage } = useProducts();
+  const onIncrement = () => {
+    if (page < totalPages) {
+      updatePage(++page);
```
Comment (ragrum) [inline src/components/Pagination/Pagination.jsx]: Prefer to use `page += 1`.

File: src/components/ProductCard/ProductCard.jsx [hunk 3]
```diff
@@ -1,21 +1,41 @@
...
+    }
+    fetchProductById(apiRef, id);
+    navigate({
+      pathname: `/product/${id}`,
+    });
+  };
   return (
     <>
-      <MyCard>
+      <MyCard onClick={() => handleClic(product.id)}>
```
Comment (ragrum) [inline src/components/ProductCard/ProductCard.jsx]: Typo `handleClic` should be `handleClick`.

**Review by ragrum — APPROVED** [review]

File: src/api/fetchData.js (hunk 1 above)
Comment (laulujan) [reply src/api/fetchData.js]: Solved on the last [commit](https://github.com/laulujan/react-apprenticeship-capstone4/pull/3/commits/ac1b3a08724a7062fb585b159efbda8325777b88).

**Pull request merged**